src/java.compiler/share/classes/javax/lang/model/element/TypeElement.java

```diff
@@ -53,6 +53,25 @@
  * source of information is Java source code, then the elements will be
  * returned in source code order.
  *
+ * @apiNote
+ * The represented class or interface may have an explicit backing
+ * construct (either {@linkplain
+ * javax.annotation.processing.Filer#createSourceFile(CharSequence,
```

Comment:
The `Filer` references here feel a bit weird. While most `TypeElement`s presumably have an explicit backing construct (i.e. a file), I suspect that only minority of the backing constructs are created using a `Filer`, but the text suggests(?) that all `TypeElement`s with a backing construct are created via `Filer`.

Reply:
In the latest pushed, pulled out the information specific to an annotation processing. New plain text:

ModuleElement:
"The represented module may have an explicit reference representation (either source code or executable output) or may be created from implicit information. The explicit and standalone source code construct for a module is typically a module-info.java file (JLS 7.7). Automatic modules (JLS 7.7.1) are named modules that do not have a module-info file. Implicit information is used to model unnamed modules.

In the context of annotation processing, a module element can be:
-     created from the initial inputs to a run of the tool
-     queried for in the configured environment 

PackageElement
The represented package may have an explicit reference representation (either source code or executable output) or may be created from implicit information. The explicit and standalone source code construct for a package is typically a package-info.java file (JLS 7.4.1). Implicit information is used to model unnamed packages.

In the context of annotation processing, a package element can be:
-     created from the initial inputs to a run of the tool
-     created from source code or class files written by a processor
-     queried for in the configured environment 

TypeElement
    The represented class or interface may have a reference representation (either source code or executable output). Multiple classes and interfaces can share the same reference representation backing construct. For example, multiple classes and interface can be declared in the same source file, including, but are not limited to:
-         a top-level class or interface and auxiliary classes and interfaces
-         a top-level class or interface and nested class and interfaces within it 

    In the context of annotation processing, a type element can be:
-         created from the initial inputs to a run of the tool
-         created from source code or class files written by a processor
-         queried for in the configured environment